Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TetrahedronFEMForceField.inl

```diff
@@ -1832,56 +1834,78 @@ void TetrahedronFEMForceField<DataTypes>::drawTrianglesFromRangeOfTetrahedra(
             p[vId] = x[(*it)[vId]];
         }
 
-        const Coord center = (p[0] + p[1] + p[2] + p[3]) * 0.125;
-
-        if ( !showWireFrame )
+        if ( !showWireFrame && d_showElementGapScale.getValue() != 0.0 )
         {
+            const Coord center = (p[0] + p[1] + p[2] + p[3]) * 0.25;
+
+            if(d_showElementGapScale.getValue() > 1.0) d_showElementGapScale.setValue(1.0);
```

Comment:
Clamping the value should be done at init() or in a specific data update callback. The clamping should be implemented that way so users knows it is clamped:
```cpp
if(d_showElementGapScale.getValue() > 1.0)
{
     d_showElementGapScale.setValue(1.0);
     msg_warning() << "the showElementGapScale is " << d_showElementGapScale.getValue() << "  but it can only be between [0.0,1.0]. Clamping to " << 1.0 << msgendl << "To remove this message you need to specify a value between 1.00 and 1.0"; 
}

```